stdlib/database/sql/src/main/java/org/ballerinalang/database/sql/SQLDatasourceUtils.java

```diff
@@ -1114,16 +1111,13 @@ public static String getString(Struct udt) throws SQLException {
         return null;
     }
 
-    public static BMap<?, ?> getSQLConnectorError(Context context, Throwable throwable) {
-        PackageInfo sqlPackageInfo = context.getProgramFile().getPackageInfo(BLangConstants.BALLERINA_BUILTIN_PKG);
-        StructureTypeInfo errorStructInfo = sqlPackageInfo.getStructInfo(Constants.SQL_CONNECTOR_ERROR);
-        BMap<String, BValue> sqlConnectorError = new BMap<>(errorStructInfo.getType());
-        if (throwable.getMessage() == null) {
-            sqlConnectorError.put(ERROR_MESSAGE_FIELD, new BString(Constants.SQL_EXCEPTION_OCCURED));
-        } else {
-            sqlConnectorError.put(ERROR_MESSAGE_FIELD, new BString(throwable.getMessage()));
-        }
-        return sqlConnectorError;
+    public static BError getSQLConnectorError(Context context, Throwable throwable) {
+        String detailedErrorMessage = throwable.getMessage() != null ? throwable.getMessage() : Constants.DATABASE_ERROR_MESSAGE;
```

Comment:
Is there any chance to throw any Java specific exception from this throwable?

Reply:
Excepted exceptions are SQLExceptions here which do not have java specific errors. Still I cannot say there is zero probability for a java specific exception to be thrown. 
However, I will be refactoring error handling code to handle only SQL errors/Connection pool errors etc and let the other exceptions be thrown with a custom message.

Reply:
Can you please check the line length here?